localstack-core/localstack/utils/analytics/usage.py

```diff
@@ -101,13 +112,11 @@ def aggregate() -> dict:
     return aggregated_payload
 
 
+@hooks.on_infra_shutdown(should_load=lambda: not config.DISABLE_EVENTS)
 def aggregate_and_send():
     """
     Aggregates data from all registered usage trackers and immediately sends the aggregated result to the analytics service.
     """
-    if config.DISABLE_EVENTS:
-        return
```

Comment:
wow thanks for adding this, can't believe i missed this :sweat_smile: 

i personally probably would have kept the guard in the method and not use `should_load`, simply because it makes it a bit easier to debug in case something's up. it also allows us to change `config.DISABLE_EVENTS` after the runtime has initiated, though we really don't have a case for that now it would be a bit more consistent with being able to set certain config values at runtime via the config endpoint.

Reply:
I see, yes, I thought about this as well! Will update. 

Only thing is that with the current logic to avoid "counting" if not enabled, even if we update at runtime, we will still avoid counting for now, or I should update the code to use the config variable directly. We can always update later for that part. 👍